tests/src/test/scala/it/unibo/scafi/test/functional/stdlib/TestGradient.scala

```diff
@@ -0,0 +1,240 @@
+package it.unibo.scafi.test.functional.stdlib
+
+import it.unibo.scafi.test.FunctionalTestIncarnation._
+import it.unibo.scafi.test.functional.{ScafiAssertions, ScafiTestUtils}
+import org.scalatest.Assertions.assert
+import org.scalatest._
+
+class TestGradient extends FunSpec with BeforeAndAfterEach {
+  import ScafiAssertions._
+  import ScafiTestUtils._
+
+  var stdNet: Network with SimulatorOps = standardNetwork()
+  val infinity: Double = Double.PositiveInfinity
+
+  def restartNetwork(): Unit = {
+    stdNet = standardNetwork()
+    stdNet.addSensor(name = "source", value = false)
+  }
+  override protected def beforeEach(): Unit = restartNetwork()
+
+  private[this] trait TestProgram extends AggregateProgram with ConstructsSemantics with StandardSensors with BlockG with StateManagement with Gradients with GenericUtils
+
+  describe("Classic Gradient") {
+    describe("On the standard network") {
+      it("Should be possible to build a gradient of distances on node 0") {
+        stdNet.chgSensorValue("source", Set(0), true)
+        exec(new TestProgram {
+          override def main(): Double = classicGradient(sense("source"))
+        }, ntimes = fewRounds)(stdNet)
+
+        assertNetworkValues((0 to 8).zip(List(
+          0.0, 1.0, 2.0,
+          1.0, 1.41, 2.41,
+          2.0, 2.42, infinity
+        )).toMap, Some((d1:Double, d2:Double) => d1===d2 +- 0.01))(stdNet)
+      }
+      it("Should be possible to build a gradient of distances on node 4") {
```

Comment:
"from node 4"

Reply:
This test case (and similarly for the other gradients) is somewhat redundant and, if I did want to do that, I'd try to avoid code repetition. What do you think?